### PR TITLE
Filling test gaps (issue 2022) - this replaces PR 1834

### PR DESCRIPTION
**Pull request by JamesWTruher**
This replaces PR 1834 from the private fork I had
it adds approximately 30 new tests for cmdlets which currently have little or no tests

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,76 @@
+Describe "Job Cmdlet Tests" -Tag "CI" {
```
Comment (daxian-dbw) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1]: job cmdlets don't work on *unix platforms, see #1972

Comment (JamesWTruher) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1]: as it turns out the cmdlets actually do run on windows - the jobs themselves don't work, but the creation/retrieval cmdlets do work. Also, currently the job creation fails on windows which is why these tests are marked as pending

Comment (daxian-dbw) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1]: thanks Jim, that sounds reasonable. BTW, what does `-pending` mean in Pester tests?

**Comment by daxian-dbw** [comment]
LGTM.

File: test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1 (hunk 1 above)
Comment (JamesWTruher) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1]: `pending` is a way that we can track those tests which will be eventually enabled. We will start tracking Pending tests and drive that count to 0 over time. A skipped test is a test which isn't really ever applicable (for example, a test that is only applicable to Linux, Core. or windows).

**Comment by JamesWTruher** [comment]
thanks!

**Pull request merged**